client/components/date-range/index.js

```diff
@@ -638,12 +658,16 @@ export class DateRange extends Component {
 		} );
 
 		const triggerProps = {
+			startDate: this.state.startDate,
+			endDate: this.state.endDate,
 			startDateText: this.toDateString( this.state.startDate ),
 			endDateText: this.toDateString( this.state.endDate ),
 			buttonRef: this.triggerButtonRef,
 			onTriggerClick: this.togglePopover,
+			onClearClick: this.clearDates,
```

Comment:
I think we may need an `onClear` prop or similar for controlled usages. If we were to connect this to redux I don't think we'd have a clean way of updating a state slice via a connected action.
I may be wrong, but I think we'd lose the ability to do what https://github.com/Automattic/wp-calypso/pull/30323/files#diff-f30509e27f891d5bbbf525f40a3e6f67R272 does here without the additional prop

Reply:
This appears to relate to the question I left in the PR description:

> Should the act of clearing the dates trigger the callback props onDateSelect and onDateCommit? If not then you need to remember to manually wire up a handler for these in the component that is consuming the DateRange component.

Is this the same issue? Ie: we need a way to propagate the clearing of state outwards?

Reply:
> Is this the same issue?

Ahh yes, I got a bit confused there but that does look to be the same issue, just in a slightly different manner. With an `onClear` prop there's a particular and explicit action taking place. By calling `onDateCommit` with `null` values (effectively the same) we're communicating up in an implicit way.
With this approach we, on the one hand, are avoiding adding another prop, but on the other hand are being less explicit - the person using this component would have to add extra checking logic in to their handler, something like so:

```js
onDateCommit = ( startDate, endDate ) => {

  if ( ! startDate && ! endDate ) {
    // infer a sort of 'clearDates' event and run relevant logic
    return;
  }
  // continue with regular onCommitDates logic
}
```

That said we might end up getting in to issues by running `onDateCommit` and `onClear` together

I'm not sure which way is best to be honest. My initial thought was the explicit `onClear` but now I'm leaning the way of just calling `onDateCommit`.

> We need a way to propagate the clearing of state outwards?

For sure, one way or another

Reply:
I've gone with triggering `onDateCommit` when the dates are cleared. Take your point about consuming requiring additional logic but I think they will just need to handle that.